src/context/HistoryContext.tsx

```diff
@@ -0,0 +1,15 @@
+import React, { createContext, useContext } from 'react';
+import type { History } from 'history';
+import history from 'history/browser';
+
+export type HistoryContextValue = History;
+
+export const HistoryContext = createContext<HistoryContextValue>(history);
+
+export const HistoryContextProvider: React.FC = ({ children }) => (
+  <HistoryContext.Provider value={history}>{children}</HistoryContext.Provider>
+);
+
+export const useHistory = () => {
```

Comment:
[IMO] A custom hook should be exposed from `src/hooks` dir.

```ts
/* src/hooks/useHistory.ts */
import { useContext } from 'react';
import { HistoryContext } from '../context/HistoryContext';

export const useHistory = () => {
  return useContext(HistoryContext);
};
```

Reply:
Fixed by https://github.com/reg-viz/reg-cli-report-ui/pull/40/commits/cea018a4a6098ed6a55e89ace832966e278eeb7c